code/ship/ship.cpp

```diff
@@ -6876,6 +6881,7 @@ void ship_subsys::clear()
 	system_info = NULL;
 
 	parent_objnum = -1;
+	parent_subsys_index = -1;
```

Comment:
While I like the idea of initializing this variable to an invalid value like -1, I don't see any places where we confirm that the value *isn't* -1. Perhaps one or more strategically placed `Assertion()`s are in order?

Reply:
I've added one.